### PR TITLE
[Feature:Developer] twig debug extension on developer environments

### PR DESCRIPTION
**Pull request by shailpatels**
Adds the the Twig DebugExtension if the debug environment is set to true, this will allow the dump function in twig files for development

### CODE REVIEW

**Comment by bmcutler** [comment]
Can we add a lint-like travis test to check for dump() in *twig and console.log() in *js?

File: site/app/libraries/Output.php [hunk 1]
```diff
@@ -78,6 +78,12 @@ public function loadTwig($full_load = true) {
             'cache' => $debug ? false : $cache_path,
             'debug' => $debug
         ]);
+
+        if($this->core->getConfig()->isDebug()){
```
Comment (MasterOdin) [inline site/app/libraries/Output.php]: Actually, now that I'm looking at this again, change this to `if ($debug)` to use the variable defined above.

**Comment by MasterOdin** [comment]
Yeah, we can add a linter for JS (like [eslint](https://eslint.org/)) to prevent things like console.log in JS files, but I'm not sure there's a good mature solution for Twig (there's [twigcs](https://github.com/allocine/twigcs), but it's pretty early in development cycle).

I could do something like Symfony itself does which is load a "production" twig environment and then just load all twig environments and if any fail, then fail. Looking into this, it looks like in production (e.g. where `debug = false`), attempting to load a template that uses `{{ dump(...) }}` will throw an exception.

**Comment by shailpatels** [comment]
> Yeah, we can add a linter for JS (like [eslint](https://eslint.org/)) to prevent things like console.log in JS files, but I'm not sure there's a good mature solution for Twig (there's [twigcs](https://github.com/allocine/twigcs), but it's pretty early in development cycle).
> 
> I could do something like Symfony itself does which is load a "production" twig environment and then just load all twig environments and if any fail, then fail. Looking into this, it looks like in production (e.g. where `debug = false`), attempting to load a template that uses `{{ dump(...) }}` will throw an exception.

Yeah if debug is set to false, {{ dump(...) }} will throw a frog robot PHP exception and I think it'll say something along the line of "unknown "dump()" function".

**Pull request merged**